completion/available/dotnet.completion.bash

```diff
@@ -2,17 +2,13 @@
 about-completion "bash parameter completion for the dotnet CLI"
 # see https://docs.microsoft.com/en-us/dotnet/core/tools/enable-tab-autocomplete#bash
 
-_dotnet_bash_complete()
-{
-  local word=${COMP_WORDS[COMP_CWORD]}
+function _dotnet_bash_complete() {
+	local cur="${COMP_WORDS[COMP_CWORD]}" IFS=$'\n'
+	local candidates
 
-  local completions
-  completions="$(dotnet complete --position "${COMP_POINT}" "${COMP_LINE}" 2>/dev/null)"
-  if [ $? -ne 0 ]; then
-    completions=""
-  fi
+	read -d '' -ra candidates < <(dotnet complete --position "${COMP_POINT}" "${COMP_LINE}" 2> /dev/null)
 
-  COMPREPLY=( $(compgen -W "$completions" -- "$word") )
+	read -d '' -ra COMPREPLY < <(compgen -W "${candidates[*]:-}" -- "$cur")
```

Comment:
```suggestion
	read -d '' -ra COMPREPLY < <(compgen -W "${candidates[*]}" -- "$cur")
```

Reply:
Prior to _Bash_ v4.4, lacking a default value produces an error if `dotnet complete` doesn't return any possibilities (e.g., if it's not installed or just there's no available completions for the provided command) when `nounset` is enabled. We support all the way back to _Bash_ v3.2 as that's the last GPLv2 release.